src/librustc_mir/interpret/validity.rs

```diff
@@ -484,8 +483,9 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                 // We are conservative with undef for integers, but try to
                 // actually enforce the strict rules for raw pointers (mostly because
                 // that lets us re-use `ref_to_mplace`).
-                let place = try_validation!(
+                let place = try_validation_pat!(
                     self.ecx.ref_to_mplace(self.ecx.read_immediate(value)?),
+                    InterpErrorInfo { kind: err_ub!(InvalidUndefBytes(..)), ..},
```

Comment:
Please adjust `try_validation_pat` so we match on the `InterpError`, no need to repeat this unwrapping of `InterpErrorInfo` everywhere.